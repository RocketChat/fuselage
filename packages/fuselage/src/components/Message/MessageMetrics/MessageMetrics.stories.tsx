import { action } from '@storybook/addon-actions';
import { Title, Primary } from '@storybook/addon-docs';
import type { ComponentMeta } from '@storybook/react';
import React from 'react';

import {
  MessageMetrics,
  MessageMetricsReply,
  MessageMetricsItem,
  MessageMetricsFollowing,
  MessageMetricsItemIcon,
  MessageMetricsItemLabel,
} from '.';
import { BasicMessageTemplate } from '../helpers';

export default {
  title: 'Message/MessageMetrics',
  component: MessageMetrics,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Primary />
        </>
      ),
    },
  },
} as ComponentMeta<typeof MessageMetrics>;

const metrics = (
  <MessageMetrics>
    <MessageMetricsReply>Reply</MessageMetricsReply>
    <MessageMetricsItem title='Replies'>
      <MessageMetricsItemIcon name='thread' />
      <MessageMetricsItemLabel>10</MessageMetricsItemLabel>
    </MessageMetricsItem>
    <MessageMetricsItem>
      <MessageMetricsFollowing
        title='Following'
        name='bell'
        onClick={action('click')}
      />
    </MessageMetricsItem>
  </MessageMetrics>
);

export const Default = BasicMessageTemplate.bind({});
Default.args = {
  metrics,
};
