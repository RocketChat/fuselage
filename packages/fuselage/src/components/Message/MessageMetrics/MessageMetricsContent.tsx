import type { HTMLAttributes } from 'react';
import React from 'react';

export const MessageMetricsContent = (
  props: HTMLAttributes<HTMLDivElement>
) => <div className='rcx-message-metrics__content' {...props} />;
