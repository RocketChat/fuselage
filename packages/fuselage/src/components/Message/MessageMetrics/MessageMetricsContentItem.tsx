import type { ReactNode } from 'react';
import React from 'react';

type MessageMetricsContentItemProps = {
  children?: ReactNode;
};

export const MessageMetricsContentItem = (
  props: MessageMetricsContentItemProps
) => <div className='rcx-message-metrics__content-item' {...props} />;
