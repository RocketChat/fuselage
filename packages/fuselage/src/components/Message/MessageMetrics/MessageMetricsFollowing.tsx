import type { ComponentProps } from 'react';
import React from 'react';

import { IconButton } from '../../Button';

type MessageMetricsFollowingProps = {
  name: 'bell' | 'bell-off';
} & Omit<ComponentProps<typeof IconButton>, 'icon'>;

export const MessageMetricsFollowing = ({
  name,
  ...props
}: MessageMetricsFollowingProps) => <IconButton {...props} small icon={name} />;
