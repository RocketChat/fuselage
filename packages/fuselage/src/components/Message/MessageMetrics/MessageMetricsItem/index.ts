import { MessageMetricsItem } from './MessageMetricsItem';
import { MessageMetricsItemIcon } from './MessageMetricsItemIcon';
import { MessageMetricsItemLabel } from './MessageMetricsItemLabel';

export default Object.assign(MessageMetricsItem, {
  Icon: MessageMetricsItemIcon,
  Label: MessageMetricsItemLabel,
});
