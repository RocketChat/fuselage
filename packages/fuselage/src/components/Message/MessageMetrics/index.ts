export {
  default,
  default as MessageMetrics,
  MessageMetricsProps,
} from './MessageMetrics';
export {
  default as MessageMetricsContentItem,
  MessageMetricsContentItemProps,
} from './MessageMetricsContentItem';
export {
  MessageMetricsItem,
  MessageMetricsItemProps,
  MessageMetricsItemIcon,
  MessageMetricsItemIconProps,
  MessageMetricsItemLabel,
  MessageMetricsItemLabelProps,
} from './MessageMetricsItem';
export {
  default as MessageMetricsFollowing,
  MessageMetricsFollowingProps,
} from './MessageMetricsFollowing';
export {
  default as MessageMetricsReply,
  MessageMetricsReplyProps,
} from './MessageMetricsReply';
