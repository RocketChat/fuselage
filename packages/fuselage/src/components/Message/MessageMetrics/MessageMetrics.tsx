import type { HTMLAttributes } from 'react';
import React from 'react';

import { MessageMetricsContentItem } from './MessageMetricsContentItem';

type MessageMetricsProps = HTMLAttributes<HTMLDivElement>;

export const MessageMetrics = (props: MessageMetricsProps) => (
  <MessageMetricsContentItem>
    <div className='rcx-message-metrics__content-wrapper' {...props} />
  </MessageMetricsContentItem>
);
