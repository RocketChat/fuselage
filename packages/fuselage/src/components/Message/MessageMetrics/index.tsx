import { MessageMetrics } from './MessageMetrics';
import { MessageMetricsFollowing } from './MessageMetricsFollowing';
import MessageMetricsItem from './MessageMetricsItem';
import { MessageMetricsReply } from './MessageMetricsReply';

export default Object.assign(MessageMetrics, {
  Reply: MessageMetricsReply,
  Item: MessageMetricsItem,
  Following: MessageMetricsFollowing,
});

export {
  MessageMetrics,
  MessageMetricsFollowing,
  MessageMetricsItem,
  MessageMetricsReply,
};
