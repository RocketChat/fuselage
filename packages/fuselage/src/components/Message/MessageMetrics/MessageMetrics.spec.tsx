import { render } from '@testing-library/react';
import React from 'react';

import MessageMetrics from '.';

it('renders without crashing', () => {
  render(
    <MessageMetrics>
      <MessageMetrics.Reply />
      <MessageMetrics.Item />
      <MessageMetrics.Following name='bell' />
    </MessageMetrics>
  );
});
