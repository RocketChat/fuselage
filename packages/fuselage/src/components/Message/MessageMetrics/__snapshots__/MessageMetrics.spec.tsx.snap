// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MessageMetrics Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full"
    >
      <div
        class="rcx-message-divider"
        role="separator"
      >
        <div
          class="rcx-message-divider__bar"
        />
        <div
          class="rcx-message-divider__wrapper"
        >
          May, 24, 2020
        </div>
         
        <div
          class="rcx-message-divider__bar"
        />
      </div>
      <div
        class="rcx-message rcx-message--clickable customclass"
        tabindex="0"
      >
        <div
          class="rcx-box rcx-box--full rcx-message-container rcx-message-container--left"
        >
          <figure
            class="rcx-box rcx-box--full rcx-avatar rcx-avatar--x36"
          >
            <img
              alt=""
              class="rcx-avatar__element rcx-avatar__element--x36"
              src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAgGBgcGBQgHBwcJCQgKDBQNDAsLDBkSEw8UHRofHh0aHBwgJC4nICIsIxwcKDcpLDAxNDQ0Hyc5PTgyPC4zNDL/2wBDAQkJCQwLDBgNDRgyIRwhMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjL/wAARCAAoACgDASIAAhEBAxEB/8QAGwAAAgIDAQAAAAAAAAAAAAAAAAcEBgIDBQj/xAAuEAACAQQAAwcEAQUAAAAAAAABAgMABAUREiExBhMUIkFRYQcWcYGhFTJSgpH/xAAYAQADAQEAAAAAAAAAAAAAAAACAwQBAP/EAB4RAAIBBQEBAQAAAAAAAAAAAAABAgMREiExE0HR/9oADAMBAAIRAxEAPwBuXuIkhBuMe5ib/AHQP49q4L3mLitryTLTSpOiHQI5k/HzXa/qbFOEudVTu1dumWvcTaNCZYZ7vU6g6LxqjOU/24dfs1Ouh9FnkMpd3Reeyx83hAxZZEhkdV9/MBrX71WGPvJcqrJBGveKATtuXXqNU0pu02bTHXD/AGvJAluyxxRd6F4x00o+NdKoVrjbzJdvVe1t5cVLc2ck8qjnohgpPtz2v7G6JtPQ2VJwjlcw+37mchpnK6GtIuv5NFWeTsLNPvxWTvpfjvOEfwKKzEVkSct2vscS/BIzSN0YRkeX81UpPqO8masJETu7OOccY4dswYFQeftv096XV5knuJGdm2T1+agvMXj8jEaHX905QihabvcbuS7X566mLWLwSY8PuRnk/u4eZ0deTl71Ef6hY+0yM88TzeNZY4luYwpVYyduOfrvhPTnr0pXSX9y5mCsyJMdyxxvwq599em+taItqCSNc90ChvZRUruUcT0JiO18Elpk7t8v41LWzacxkBSuvjQ/FFJayjDWrCTepAQ2vUH0oo/Jk3ovpwJJeVCP5CN+lFFaaMqy+nAyuChvrTI2kN9JAsi2ZOy4IBHMnkSCP+iqBexSWdxLazoUljJVlPUH2oorkV10pRc7b1zXb/hZOzuJvM86QWEXeELxOzHSIPcmiiiunVlF2RNTpRkrs//Z"
            />
          </figure>
        </div>
        <div
          class="rcx-box rcx-box--full rcx-message-container"
        >
          <div
            class="rcx-box rcx-box--full rcx-message-header"
          >
            <div
              class="rcx-box rcx-box--full rcx-message-header__wrapper"
            >
              <span
                class="rcx-box rcx-box--full rcx-message-header__name-container"
              >
                <span
                  class="rcx-box rcx-box--full rcx-message-header__name"
                >
                  Haylie George
                </span>
                 
                <span
                  class="rcx-box rcx-box--full rcx-message-header__username"
                >
                  @haylie.george
                </span>
              </span>
              <div
                class="rcx-box rcx-box--full rcx-message-header__roles"
              >
                <span
                  class="rcx-box rcx-box--full  rcx-tag  rcx-box rcx-box--full rcx-message-header__role"
                >
                  <span
                    class="rcx-tag__inner"
                  >
                    Admin
                  </span>
                </span>
                <span
                  class="rcx-box rcx-box--full  rcx-tag  rcx-box rcx-box--full rcx-message-header__role"
                >
                  <span
                    class="rcx-tag__inner"
                  >
                    User
                  </span>
                </span>
                <span
                  class="rcx-box rcx-box--full  rcx-tag  rcx-box rcx-box--full rcx-message-header__role"
                >
                  <span
                    class="rcx-tag__inner"
                  >
                    Owner
                  </span>
                </span>
              </div>
              <span
                class="rcx-box rcx-box--full rcx-message-header__time"
              >
                12:00 PM
              </span>
            </div>
          </div>
          <div
            class="rcx-message-body"
          >
            Ut enim ad minim veniam
          </div>
          <div
            class="rcx-box rcx-box--full rcx-message-block"
          >
            <div
              class="rcx-message-metrics__content-item"
            >
              <div
                class="rcx-message-metrics__content-wrapper"
              >
                <div
                  class="rcx-message-metrics__item"
                >
                  <button
                    class="rcx-box rcx-box--full rcx-button--small rcx-button--primary rcx-button"
                    type="button"
                  >
                    <span
                      class="rcx-button--content"
                    >
                      Reply
                    </span>
                  </button>
                </div>
                <div
                  class="rcx-message-metrics__item"
                  title="Replies"
                >
                  <i
                    aria-hidden="true"
                    class="rcx-box rcx-box--full rcx-icon--name-thread rcx-icon rcx-css-4pvxx3"
                  >
                    
                  </i>
                  <div
                    class="rcx-message-metrics__item-label"
                  >
                    10
                  </div>
                </div>
                <div
                  class="rcx-message-metrics__item"
                >
                  <button
                    class="rcx-box rcx-box--full rcx-button--small-square rcx-button--square rcx-button--icon rcx-button"
                    title="Following"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      class="rcx-box rcx-box--full rcx-icon--name-bell rcx-icon rcx-css-4pvxx3"
                    >
                      
                    </i>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
